esp-rom-sys/src/syscall/mod.rs

```diff
@@ -0,0 +1,148 @@
+            _abort: Some(abort_wrapper),
+            _system_r: Some(core::mem::transmute(not_implemented as usize)),
+            _rename_r: Some(core::mem::transmute(not_implemented as usize)),
+            _times_r: Some(core::mem::transmute(not_implemented as usize)),
+            _gettimeofday_r: Some(core::mem::transmute(not_implemented as usize)),
+            _raise_r: Some(core::mem::transmute(not_implemented as usize)),
+            _unlink_r: Some(core::mem::transmute(not_implemented as usize)),
+            _link_r: Some(core::mem::transmute(not_implemented as usize)),
+            _stat_r: Some(core::mem::transmute(not_implemented as usize)),
+            _fstat_r: Some(core::mem::transmute(not_implemented as usize)),
+            _sbrk_r: Some(core::mem::transmute(not_implemented as usize)),
+            _getpid_r: Some(core::mem::transmute(not_implemented as usize)),
+            _kill_r: Some(core::mem::transmute(not_implemented as usize)),
+            _exit_r: Some(core::mem::transmute(not_implemented as usize)),
+            _close_r: Some(core::mem::transmute(not_implemented as usize)),
+            _open_r: Some(core::mem::transmute(not_implemented as usize)),
+            _write_r: Some(core::mem::transmute(not_implemented as usize)),
+            _lseek_r: Some(core::mem::transmute(not_implemented as usize)),
+            _read_r: Some(core::mem::transmute(not_implemented as usize)),
+            _retarget_lock_init: Some(core::mem::transmute(not_implemented as usize)),
+            _retarget_lock_init_recursive: Some(core::mem::transmute(not_implemented as usize)),
+            _retarget_lock_close: Some(core::mem::transmute(not_implemented as usize)),
+            _retarget_lock_close_recursive: Some(core::mem::transmute(not_implemented as usize)),
+            _retarget_lock_acquire: Some(core::mem::transmute(not_implemented as usize)),
+            _retarget_lock_acquire_recursive: Some(core::mem::transmute(not_implemented as usize)),
+            _retarget_lock_try_acquire: Some(core::mem::transmute(not_implemented as usize)),
+            _retarget_lock_try_acquire_recursive: Some(core::mem::transmute(
+                not_implemented as usize,
+            )),
+            _retarget_lock_release: Some(core::mem::transmute(not_implemented as usize)),
+            _retarget_lock_release_recursive: Some(core::mem::transmute(not_implemented as usize)),
+            _printf_float: Some(core::mem::transmute(not_implemented as usize)),
+            _scanf_float: Some(core::mem::transmute(not_implemented as usize)),
+            __assert_func: Some(super::__assert_func),
+            __sinit: Some(core::mem::transmute(not_implemented as usize)),
+            _cleanup_r: Some(core::mem::transmute(not_implemented as usize)),
+        });
+
+        S_STUB_TABLE.assume_init();
+    };
+
+    let syscall_table =
+        core::ptr::addr_of!(S_STUB_TABLE) as *const chip_specific::syscall_stub_table;
```

Comment:
MaybeUninit::write returns a mutable reference to its data, you could cast that into a pointer.